### PR TITLE
Updated Code Examples in Integration Docs for v3 Mail Send

### PR DESCRIPTION
**Pull request by eschallock**
## Update Code Examples for v3 Mail Send
* Created /Integrate/Code_Examples/v2_Mail
* Created /Integrate/Code_Examples/v3_Mail
* /Integrate/Code_Examples/index.html
 * Updated to contain info for both v2 and v3 examples
* /Integrate/Code_Examples/v2_Mail/index.html
 * Created new index for v2_Mail code examples
* /Integrate/Code_Examples/v3_Mail/index.html
 * Created new index for v3_Mail code examples
* /_data/folder_names.yml
 * v3_Mail is named "v3 Mail Beta" temporarily
* /_redirects.htaccess
 * Added redirects for v2 examples that were moved to v2_Mail

### CODE REVIEW

**Comment by eschallock** [comment]
@thinkingserious Would you be able to quickly review these changes before I make the merge? Specifically, if you could double check the added gists in /Integrate/Code_Examples/v3_Mail/.

File: source/Integrate/Code_Examples/index.html [hunk 1]
```diff
@@ -7,54 +7,42 @@
...
+  </div>
+  <div class="col-md-4 subsection-links">
+    <br />
+    <p class="subsection-title">Libraries</p>
+      <p><a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/csharp.html">C#<span class="icon-net" alt="C#"></span></a><br />
+      <a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/go.html">Go<br />
+      <a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/java.html">Java<span class="icon-java"></span></a><br />
+      <a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/nodejs.html">Node.js<span class="icon-nodejs"></span></a><br />
+      <a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/php.html">PHP<span class="icon-php"></span></a><br />
+      <a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/python.html">Pyton<span class="icon-python"></span></a><br />
```
Comment (thinkingserious) [inline source/Integrate/Code_Examples/index.html]: Pyton -> Python

File: source/Integrate/Code_Examples/index.html [hunk 2]
```diff
@@ -7,54 +7,42 @@
...
+      <h2><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/">Web API v3 Code Examples</a></h2>
+      <p>
+      Learn how to quickly start sending email through SendGrid's Web API v3. Here you will find code examples for our latest v3 Mail Send endpoint in 7 different languages.
+      </p>
+  </div>
+  <div class="col-md-4 subsection-links">
+    <br />
+    <p class="subsection-title">Libraries</p>
+      <p><a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/csharp.html">C#<span class="icon-net" alt="C#"></span></a><br />
+      <a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/go.html">Go<br />
```
Comment (thinkingserious) [inline source/Integrate/Code_Examples/index.html]: Would be cool to get a Go icon: https://www.google.com/search?q=golang+icon&biw=1882&bih=1323&tbm=isch&tbo=u&source=univ&sa=X&ved=0ahUKEwj8_tbIot_MAhUO3mMKHaeTD1sQsAQIGw

File: source/Integrate/Code_Examples/index.html [hunk 3]
```diff
@@ -7,54 +7,42 @@
...
+      <a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/python.html">Pyton<span class="icon-python"></span></a><br />
+      <a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/ruby.html">Ruby<span class="icon-ruby"></span></a></p>
+  </div>
+</div>
+
+<div class="row">
+  <div class="col-md-8">
+      <h2><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/">Web API v3 Code Examples</a></h2>
+      <p>
+      Learn how to quickly start sending email through SendGrid's Web API v3. Here you will find code examples for our latest v3 Mail Send endpoint in 7 different languages.
```
Comment (thinkingserious) [inline source/Integrate/Code_Examples/index.html]: language -> programming languages

File: source/Integrate/Code_Examples/v2_Mail/index.html [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+---
+
+<ul>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/csharp.html">C#<span class="icon-net" alt="C#"></span></a><br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/go.html">Go<br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/java.html">Java<span class="icon-java"></span></a><br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/nodejs.html">Node.js<span class="icon-nodejs"></span></a><br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/perl.html">Node.js<span class="icon-perl"></span></a><br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/php.html">PHP<span class="icon-php"></span></a><br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/python.html">Pyton<span class="icon-python"></span></a><br /></li>
```
Comment (thinkingserious) [inline source/Integrate/Code_Examples/v2_Mail/index.html]: Pyton -> Python

File: source/Integrate/Code_Examples/v2_Mail/index.html [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+navigation:
+  show: true
+---
+
+<ul>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/csharp.html">C#<span class="icon-net" alt="C#"></span></a><br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/go.html">Go<br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/java.html">Java<span class="icon-java"></span></a><br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/nodejs.html">Node.js<span class="icon-nodejs"></span></a><br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/perl.html">Node.js<span class="icon-perl"></span></a><br /></li>
```
Comment (thinkingserious) [inline source/Integrate/Code_Examples/v2_Mail/index.html]: Should be Perl

File: source/Integrate/Code_Examples/v2_Mail/index.html [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+layout: page
+weight: 100
+title: v2 Code Examples Index
+navigation:
+  show: true
+---
+
+<ul>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/csharp.html">C#<span class="icon-net" alt="C#"></span></a><br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v2_Mail/go.html">Go<br /></li>
```
Comment (thinkingserious) [inline source/Integrate/Code_Examples/v2_Mail/index.html]: If you get a Go icon, here is another place to put it

File: source/Integrate/Code_Examples/v3_Mail/index.html [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  show: true
+---
+
+<ul>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/csharp.html">C#<span class="icon-net" alt="C#"></span></a><br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/go.html">Go<br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/java.html">Java<span class="icon-java"></span></a><br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/nodejs.html">Node.js<span class="icon-nodejs"></span></a><br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/php.html">PHP<span class="icon-php"></span></a><br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/python.html">Pyton<span class="icon-python"></span></a><br /></li>
```
Comment (thinkingserious) [inline source/Integrate/Code_Examples/v3_Mail/index.html]: Pyton ->Python

File: source/Integrate/Code_Examples/v3_Mail/index.html [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+layout: page
+weight: 100
+title: v3 Code Examples Index
+navigation:
+  show: true
+---
+
+<ul>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/csharp.html">C#<span class="icon-net" alt="C#"></span></a><br /></li>
+  <li><a href="{{root_url}}/Integrate/Code_Examples/v3_Mail/go.html">Go<br /></li>
```
Comment (thinkingserious) [inline source/Integrate/Code_Examples/v3_Mail/index.html]: Another place for a shiny new Go icon

**Pull request merged**